server/embed/etcd.go

```diff
@@ -774,3 +797,29 @@ func parseCompactionRetention(mode, retention string) (ret time.Duration, err er
 	}
 	return ret, nil
 }
+
+func (e *Etcd) setupTracing(ctx context.Context) (tracesdk.SpanExporter, error) {
+	exporter, err := otlp.NewExporter(ctx,
+		otlpgrpc.NewDriver(
+			otlpgrpc.WithEndpoint(e.cfg.ExperimentalDistributedTracingAddress),
+			otlpgrpc.WithInsecure(),
+		))
+	if err != nil {
+		return nil, err
+	}
+	res := resource.NewWithAttributes(
+		semconv.ServiceNameKey.String(e.cfg.ExperimentalDistributedTracingServiceName),
+	)
+	tp := tracesdk.NewTracerProvider(
+		tracesdk.WithBatcher(exporter),
+		tracesdk.WithResource(res),
+	)
+	otel.SetTracerProvider(tp)
```

Comment:
in addition to setting the global TracerProvider, you should also set the global propagators:

```golang
otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
```

The TraceContext propagator will allow linking traces from the APIServer to etcd, and Baggage allows passing additional tags via context.